### ISSUE TITLE
DXR raytracing examples FallbackLayer exception on third call into CreatePSOHelper (util.h) - with workaround compiles/runs, but without fallback raytracing (see picture)

### ISSUE THREAD

**Issue by CGAndersonSea** [Microsoft/DirectX-Graphics-Samples#443]
When trying to run D3D12RaytracingHelloWorld, happens in CreatePSOHelper in util.h
Exception thrown at 0x00007FFEBA765899 (amdxc64.dll) in D3D12RaytracingHelloWorld.exe: 0xC0000005: Access violation reading location 0x0000000000000028. occurred

When trying to run D3D12RaytracingSimpleLighting happens in CreatePSOHelper  in util.h

it's always because of the third call in treeletreorder.cpp (third call, line 35)
 CreatePSOHelper(pDevice, nodeMask, m_pRootSignature, COMPILED_SHADER(g_pClearBuffers), &m_pClearBuffersPSO);
        CreatePSOHelper(pDevice, nodeMask, m_pRootSignature, COMPILED_SHADER(g_pFindTreelets), &m_pFindTreeletsPSO);

---->this one---->        CreatePSOHelper(pDevice, nodeMask, m_pRootSignature, COMPILED_SHADER(g_pTreeletReorder), &m_pTreeletReorderPSO);

Other DirectX 12 samples (just not raytracing ones) work fine.
Card is an AMD Radeon Vega 64.  Latest public driver. Windows 10, correct SDK (latest), VS 2017. 

Made a zip of the samples (not a clone) and got the binaries and followed instructions. Everything compiles fine. Am using the desktop samples. It's only during running and trying to use the Fallback Layer that there is a problem. (Vega does not have DXR native driver, so fallback should work. that is what the code is selecting.)

Any advice on what may be causing this? Why the third call into CreatePSOHelper?

**Comment by pkristof** [comment]
Hello,

Thank you for reporting the issue. The treelet reorder has a few kinks we are ironing out on AMD in our develop branch and it seems to have trickled into master branch inadvertently. There is a macro override to disable treelet reorder in the develop branch. You should be able to unblock yourself by applying that same change (see a link to the change below) and setting the macros to 0. Let us know if that doesn't help and we'll take a look.
https://github.com/Microsoft/DirectX-Graphics-Samples/commit/a86a3903faa28427c5bbed34a23074e3af974e3f

Peter

**Comment by CGAndersonSea** [comment]
Thanks, it does let the samples run without exceptions. And it does make the miniengine sample run correctly. 

Two of the other samples 'sort of' run. they don't exception out, but they don't raytrace either - just a black area where one should see raytracing.  The third, procedural geometry, still exceptions out (DeviceResources.cpp during DeviceResources::ExecuteCommandList() during the m_commandQueue->ExecuteCommandLists(..) call.

Is that expected behavior? Do the these three other samples not fall back to the fallback layer? Or are there other AMD workarounds that need to be done to get the other samples to run correctly...or is that simply not possible at this time?  

Pending your response, I'll wait until I close this issue (or reopen another against the other three samples, if you want). Thanks!

**Comment by pkristof** [comment]
Hi,

ProceduralGeometry sample is incompatible on AMD, as per https://github.com/Microsoft/DirectX-Graphics-Samples/tree/master/Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingProceduralGeometry#known-limitations. AMD has been looking into the issue.

The two other samples are expected to run as is. We'll double check that nothing regressed after our last push to Master.

Thanks,
Peter

**Comment by CGAndersonSea** [comment]
Thanks!

**Comment by CGAndersonSea** [comment]
Just to be sure, I backed up one driver version on the Vega. Still need the treelet workaround to compile (just to confirm), and still no actual fallback raytracing when they do compile. see image of hello world, which according to you, should work. but as you can see, no raytrace image in the center. 
I double checked everything I could think of. Latest Windows update (april), installed win 10 sdk, then installed vs2017 over that, redownloaded and rebuilt everything, and so on.
![image](https://user-images.githubusercontent.com/41025371/44557711-c577e480-a6f4-11e8-8fc2-a3ab6642d0ce.png)

**Comment by pkristof** [comment]
Thanks for the follow up. I have reproduced the issue and indeed AMD GPU has issue raytracing for the samples as Closest Hit and Miss shaders don't update the color values. We're working with AMD on the issue. For now, I have added link to the previous source and overlay binary versions for which the samples render correctly on AMD to https://github.com/Microsoft/DirectX-Graphics-Samples/blob/master/Samples/Desktop/D3D12Raytracing/readme.md#known-limitations.
